src/Type/Constant/ConstantArrayType.php

```diff
@@ -734,7 +738,9 @@ public function unsetOffset(Type $offsetType): Type
 					$k++;
 				}
 
-				return new self($newKeyTypes, $newValueTypes, $this->nextAutoIndexes, $newOptionalKeys, TrinaryLogic::createNo());
+				$isList = $this->isList()->and(TrinaryLogic::createFromBoolean($i === count($this->keyTypes) - 1));
```

Comment:
A list stops being a list even if the last index is removed. One thing that makes list a list is what happens when you append a new item to it.

Reply:
This is untrue according to https://onlinephp.io/c/28b76
`array_is_list` still returns True after we unset a value.

Without this fix, we get an error https://github.com/phpstan/phpstan-src/actions/runs/10653522924/job/29528721910?pr=3381

```
Parameter #2 $expectedErrors of method
PHPStan\Testing\RuleTestCase<PHPStan\Rules\Comparison\ImpossibleCheckTypeFunctionCallRule>::analyse()  
expects list<array{0: string, 1: int, 2?: string|null}>,
list<array{0: string, 1: int, 2?: string}> given.        
```